server/etcdmain/grpc_proxy_logger.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2025 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package etcdmain
+
+import (
+	"context"
+	"fmt"
+	"path"
+	"reflect"
+	"slices"
+	"time"
+
+	"github.com/golang/protobuf/jsonpb"
+	"github.com/golang/protobuf/proto"
+	"github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors"
+	"go.uber.org/zap"
+	"google.golang.org/grpc/peer"
+)
+
+// proxylogger implements the go-grpc-middleware v2 Reporter interface
+type proxylogger struct {
```

Comment:
Should this be named `grcpProxyLogger` to match the filename, and to make it less generic in the etcdmain package?

Reply:
That is a more proper name.  I'm always bad at naming things :D